src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/ISearchParameterUtilities.cs

```diff
@@ -14,5 +14,7 @@ public interface ISearchParameterUtilities
         Task AddSearchParameterAsync(ITypedElement searchParam);
 
         Task DeleteSearchParameterAsync(RawResource searchParam);
+
+        Task UpdateSearchParameterAsync(ITypedElement searchParam, RawResource previousSearchParam);
```

Comment:
nit: `ISearchParameterUtilities` has always been a little bit too generic of a name for me, is it closer to a Repository/DataStore/Facade?

Reply:
Agreed.  Essentially this class coordinates calls to both the SearchParameterDefinitionManager and SearchParameterStatusRegistry when changes need to happen.  It also builds proper OperationOutcomeIssues when errors occur.  I'll try to think of a better name :).